ui/src/parts/lightbox.tsx

```diff
@@ -0,0 +1,224 @@
+      textAlign: 'center',
+      color: '#fff',
+    },
+    imageNav: {
+      height: IMAGE_NAV_HEIGHT,
+      overflow: 'auto',
+      whiteSpace: 'nowrap'
+    },
+    navImg: {
+      boxSizing: 'border-box',
+      objectFit: 'cover',
+      height: '120px',
+      width: '120px',
+      margin: '0px 2px',
+      cursor: 'pointer',
+      filter: 'brightness(30%)',
+      border: '2px solid #000'
+    },
+    imgCaptions: {
+      whiteSpace: 'nowrap',
+      padding: '10px 40px',
+      height: IMAGE_CAPTIONS_HEIGHT - 20
+    },
+    text: {
+      textOverflow: 'ellipsis',
+      overflow: 'hidden'
+    },
+    title: {
+      fontWeight: 500
+    },
+    description: {
+      color: '#bbb'
+    },
+    navImgContainer: {
+      display: 'inline-block',
+      width: '124px'
+    },
+    iconStylesRootArrow: {
+      position: "absolute",
+      top: "50%",
+      padding: "16px",
+      marginTop: "-50px"
+    }
+  }),
+  styles: { [key: S]: React.CSSProperties } = {
+    icon: { fontSize: '22px' }
+  },
+  iconStyles: Fluent.IButtonStyles = {
+    flexContainer: { justifyContent: 'center' },
+    root: { margin: '4px 4px', width: 38, height: 38, backgroundColor: 'rgba(0, 0, 0, 0.3)' },
+    rootHovered: { backgroundColor: 'rgba(255, 255, 255, 0.3)' },
+    icon: { color: '#fff', lineHeight: 22, height: 'unset', padding: 4 },
+    iconHovered: { color: '#fff' },
+    iconPressed: { color: 'rgba(255, 255, 255, 0.7)' },
+  }
+
+type Image = { title: S, description?: S, type?: S, image?: S, path?: S }
+
+export interface LightboxProps {
+  images: Image[],
+  defaultImageIdx?: U
+}
+
+export const Lightbox = ({ images, defaultImageIdx }: LightboxProps) => {
+  const
+    [activeImageIdx, setActiveImageIdx] = React.useState(defaultImageIdx || 0),
+    { title, description } = images[activeImageIdx],
+    isGallery = images.length > 1,
+    FOOTER_HEIGHT = isGallery ? IMAGE_CAPTIONS_HEIGHT + IMAGE_NAV_HEIGHT : IMAGE_CAPTIONS_HEIGHT,
+    imageNavRef = React.useRef<HTMLDivElement | null>(null),
+    defaultScrollSetRef = React.useRef(false),
+    activeColor = isDark(getColorFromString(cssVar('$neutralPrimary'))!) ? cssVar('$card') : cssVar('$neutralPrimary'),
+    imageHighlightStyle = { filter: 'unset', border: '2px solid', borderColor: activeColor },
+    lazyImageObserver = new window.IntersectionObserver(entries =>
+      entries.forEach(entry => {
+        if (entry.isIntersecting) {
+          const lazyImage = entry.target as HTMLImageElement
+          lazyImage.src = lazyImage.dataset.src!
+          lazyImage.classList.remove("lazy")
+          lazyImageObserver.unobserve(lazyImage)
+        }
+      })
+    ),
+    handleKeyDown = (ev: KeyboardEvent) => {
+      if (['Escape', 'ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown'].indexOf(ev.key) >= 0) ev.preventDefault()
+      if (ev.key === 'Escape') lightboxB(null)
+      if (ev.key === 'ArrowRight') setActiveImageIdx(prevIdx => prevIdx === images.length - 1 ? 0 : prevIdx + 1)
```

Comment:
We can probably use `else if` here, right?
```suggestion
      else if (ev.key === 'ArrowRight') setActiveImageIdx(prevIdx => prevIdx === images.length - 1 ? 0 : prevIdx + 1)
```